driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -96,28 +96,35 @@ fn serialize_order(order: &models::Order, id: &str) -> serde_json::Value {
     })
 }
 
+fn parse_token(key: &str) -> Result<u16, PriceFindingError> {
+    // TODO: Check that key is long enough or key[5..] wil panic.
+    key[5..]
+        .parse::<u16>()
+        .map_err(|_| PriceFindingError::new("Failed to parse token id", ErrorKind::JsonError))
+}
+
+fn parse_price_value(value: &serde_json::Value) -> Result<u128, PriceFindingError> {
+    value
+        .as_str()
+        .ok_or(PriceFindingError::from("Price value not a string"))?
+        .parse::<u128>()
+        .map_err(|_| PriceFindingError::new("Failed to parse price string", ErrorKind::JsonError))
```

Comment:
Maybe include the `ParseIntError` here so we have more details about the failure.

Reply:
How does one "include" `ParseIntError`? Do you mean to simply use the `ParseIntError` instead or like format the string with `|err| format!("{}: Failed to parse price string", err.type)`?

Reply:
I like this suggestion and will address it in the main branch! For now I will simply merge these great suggestions and continue from here! Thanks @e00E